asset/asset.go

```diff
@@ -808,6 +808,45 @@ func (a *Asset) IsUnSpendable() bool {
 		a.Amount == 0
 }
 
+// IsBurn returns true if an asset uses an un-spendable script key that was
+// constructed using the proof-of-burn scheme.
+func (a *Asset) IsBurn() bool {
+	// If the script key is nil, then we can't say if this is a burn or not.
+	if a.ScriptKey.PubKey == nil {
+		return false
+	}
+
+	var prevID PrevID
+
+	// If this is a split output, then we need to look up the first PrevID
+	// in the split root asset.
+	if a.HasSplitCommitmentWitness() {
+		if len(a.PrevWitnesses) == 0 ||
+			a.PrevWitnesses[0].SplitCommitment == nil {
+
+			return false
+		}
+
+		rootAsset := a.PrevWitnesses[0].SplitCommitment.RootAsset
+		if len(rootAsset.PrevWitnesses) == 0 ||
```

Comment:
This is just to avoid a panic if we ever get an incorrectly constructed proof. I guess we would always validate a proof first, but I'd say better safe than sorry?